Lib/test/test_statistics.py

```diff
@@ -1454,6 +1496,9 @@ class TestHarmonicMean(NumericTestCase, AverageMixin, UnivariateTypeMixin):
     def setUp(self):
         self.func = statistics.harmonic_mean
 
+    def test_single_value_unsupported_type(self):
```

Comment:
This should come after the `prepare_` methods.

Also, why test only a an array with a single value of an unsupported type? This would be a good opportunity for other unsupported cases, such as longer arrays, and arrays with values of mixed types, some supported and some not.

Reply:
You are right, I will put it right after the `test_negative_error` test.

I tested it with a single value of an unsupported type because the statement that explicitly raises this `TypeError` is part of the `elif n == 1` branch, where n is the length of the given list. So a longer list would not trigger this specific `TypeError`, which was the only statement without being covered in this function. [[1]](https://codecov.io/gh/python/cpython/src/8aea4b3605059e243f1827d9328d6fc8d698c0a7/Lib/statistics.py#L404)

If you think that it would be useful to have a test with longer lists (only with unsupported values, both supported/unsupported values) and assert that a `TypeError` is raised, I could add one in the next commit. However, this would be for a `TypeError` that is not explicitly raised by this function.

Reply:
I suggest adding two assertions:
1. multiple string values
2. mixed strings and valid values (i.e. positive numbers).

Reply:
Thanks for the suggestions @taleinat. I have added a test in the latest commit that asserts those cases.